File: Domain/Entities/Exercise.cs
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;
using Domain.Models;

namespace Domain.Entities
{
    [Table("Exercises")]
    public class Exercise : Entity<int>
    {
        public required string Name { get; set; }
        public string? Description { get; set; }
        public MuscleSubgroup MuscleSubgroup { get; set; }
        public string? ImageUrl { get; set; }
        public virtual ICollection<WorkoutExercise> WorkoutExercise { get; set; } = [];
    }
}

File: UI/src/app/core/models/Exercise.ts
import { MuscleGroup, MuscleSubgroup } from './Enums';

export interface Exercise {
  id: number;
  name: string | null;
  description: string | null;
  muscleGroup: MuscleGroup;
  muscleSubgroup: MuscleSubgroup;
  imageUrl: string | null;
}

export interface ExerciseCreate {
  name: string;
  description: string | null;
  muscleSubgroup: MuscleSubgroup;
  imageUrl: string | null;
}
